File: src/Model/ReturnValue.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System.Collections.Generic;

namespace AutoRest.Java.Model
{
    /// <summary>
    /// The details of a value that is returned from a REST API method.
    /// </summary>
    public class ReturnValue
    {
        /// <summary>
        /// Create a new ReturnValue with the provided properties.
        /// </summary>
        /// <param name="type">The type of this return value.</param>
        public ReturnValue(IType type)
        {
            Type = type;
        }

        /// <summary>
        /// The type of this return value.
        /// </summary>
        public IType Type { get; }

        /// <summary>
        /// Add this property's imports to the provided ISet of imports.
        /// </summary>
        /// <param name="imports">The set of imports to add to.</param>
        /// <param name="includeImplementationImports">Whether or not to include imports that are only necessary for method implementations.</param>
        public void AddImportsTo(ISet<string> imports, bool includeImplementationImports, JavaSettings settings)
        {
            Type.AddImportsTo(imports, includeImplementationImports);
        }
    }
}

File: mapper/src/main/java/com/azure/autorest/models/clientmodel/ReturnValue.java
package com.azure.autorest.models.clientmodel;

import java.util.*;

// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.


/** 
 A return value from a ClientMethod.
*/
public class ReturnValue
{
	/** 
	 Create a new ReturnValue object from the provided properties.
	 
	 @param description The description of the return value.
	 @param type The type of the return value.
	*/
	public ReturnValue(String description, IType type)
	{
		Description = description;
		Type = type;
	}

	/** 
	 The description of the return value.
	*/
	private String Description;
	public final String getDescription()
	{
		return Description;
	}

	/** 
	 The type of the return value.
	*/
	private IType Type;
	public final IType getType()
	{
		return Type;
	}

	/** 
	 Add this return value's imports to the provided ISet of imports.
	 
	 @param imports The set of imports to add to.
	 @param includeImplementationImports Whether or not to include imports that are only necessary for method implementations.
	*/
	public final void AddImportsTo(Set<String> imports, boolean includeImplementationImports)
	{
		getType().AddImportsTo(imports, includeImplementationImports);
	}
}